File: Stripe/STPPaymentMethodOXXO.m
//
//  STPPaymentMethodOXXO.m
//  StripeiOS
//
//  Created by Polo Li on 6/15/20.
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

#import "STPPaymentMethodOXXO.h"

#import "NSDictionary+Stripe.h"

NS_ASSUME_NONNULL_BEGIN

@implementation STPPaymentMethodOXXO

@synthesize allResponseFields = _allResponseFields;

#pragma mark - Description

- (NSString *)description {
    NSArray *props = @[
                       // Object
                       [NSString stringWithFormat:@"%@: %p", NSStringFromClass([self class]), self],
                       ];

    return [NSString stringWithFormat:@"<%@>", [props componentsJoinedByString:@"; "]];
}

#pragma mark - STPAPIResponseDecodable

+ (nullable instancetype)decodedObjectFromAPIResponse:(nullable NSDictionary *)response {
    NSDictionary *dict = [response stp_dictionaryByRemovingNulls];
    if (!dict) {
        return nil;
    }
    return [[self alloc] initWithDictionary:dict];
}

- (nullable instancetype)initWithDictionary:(NSDictionary *)dict {
    self = [super init];
    if (self) {
        _allResponseFields = dict.copy;
    }
    return self;
}

@end

NS_ASSUME_NONNULL_END

File: Stripe/STPPaymentMethodOXXO.swift
//
//  STPPaymentMethodOXXO.swift
//  Stripe
//
//  Created by Polo Li on 6/15/20.
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

import Foundation

/// An OXXO Payment Method.
/// - seealso: https://stripe.com/docs/payments/oxxo
public class STPPaymentMethodOXXO: NSObject, STPAPIResponseDecodable {
  @objc public private(set) var allResponseFields: [AnyHashable: Any] = [:]

  /// :nodoc:
  @objc public override var description: String {
    let props = [
      // Object
      String(format: "%@: %p", NSStringFromClass(STPPaymentMethodOXXO.self), self),
    ]

    return "<\(props.joined(separator: "; "))>"
  }

  // MARK: - STPAPIResponseDecodable
  override required init() {
    super.init()
  }

  @objc
  public class func decodedObject(fromAPIResponse response: [AnyHashable: Any]?) -> Self? {
    guard let response = response else {
      return nil
    }
    let oxxo = self.init()
    oxxo.allResponseFields = response
    return oxxo
  }
}
